src/ucp/wireup/select.c

```diff
@@ -695,13 +703,13 @@ ucp_wireup_add_memaccess_lanes(const ucp_wireup_select_params_t *select_params,
     status = UCS_OK;
 
 out:
-    if ((status != UCS_OK) && select_params->allow_am) {
+    if ((status != UCS_OK) && !has_reg_lane && select_params->allow_am) {
         /* using emulation over active messages */
         select_ctx->ucp_ep_init_flags |= UCP_EP_INIT_CREATE_AM_LANE;
-        status                         = UCS_OK;
+        return UCS_OK;
+    } else {
```

Comment:
[minor] can remove `else` and just `return status;` after `if (...) { .. }` construction